tools/python_api/src_cpp/pandas/pandas_scan.cpp

```diff
@@ -32,10 +33,12 @@ std::unique_ptr<TableFuncBindData> bindFunc(ClientContext* /*context*/,
     auto getFunc = df.attr("__getitem__");
     auto numRows = py::len(getFunc(columns[0]));
     auto returnColumns = input->binder->createVariables(names, returnTypes);
-    auto scanConfig =
-        input->extraInput->constPtrCast<ExtraScanTableFuncBindInput>()->fileScanInfo.copy();
-    return std::make_unique<PandasScanFunctionData>(std::move(returnColumns), df, numRows,
-        std::move(columnBindData), std::move(scanConfig));
+    auto scanConfig = PyScanConfig{
+        input->extraInput->constPtrCast<ExtraScanTableFuncBindInput>()->fileScanInfo.options};
+    KU_ASSERT(numRows >= scanConfig.skipNum);
```

Comment:
hmm Do we have a check somewhere to ensure `scanConfig.skipNum` is always smaller than `numRows`? I'm concerned if this will lead to overflow under the case `numRows < scanConfig.skipNum`. 

I would also add a test case on that if we don't have one already.